src/platform/Ameba/DiagnosticDataProviderImpl.cpp

```diff
@@ -158,6 +158,14 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface **
                 ifp->hardwareAddress = ByteSpan(ifp->MacAddress, 6);
             }
 
+            memcpy(ifp->Ipv4AddressesBuffer[0], &(ifa->ip_addr.u_addr.ip4.addr), kMaxIPv4AddrSize);
```

Comment:
is it always guaranteed that both IPv4 and IPv6 addresses area available and valid on these interfaces?

Reply:
I have added checks for the ipv4 and ipv6 addresses.